### ISSUE TITLE
StreamFailureError for SELECT 1 query

### ISSUE THREAD

**Issue by denisov-vlad** [ClickHouse/clickhouse-connect#191]
### Describe the bug

Faced the error below during the simple request. The ClickHouse server is behind `chproxy`.

```
---------------------------------------------------------------------------
StreamFailureError                        Traceback (most recent call last)
Cell In[19], line 1
----> 1 client.query('SELECT 1 as check')

File ~/Dev/streamlit-apps/.venv/lib/python3.10/site-packages/clickhouse_connect/driver/client.py:182, in Client.query(self, query, parameters, settings, query_formats, column_formats, encoding, use_none, column_oriented, use_numpy, max_str_len, context, query_tz, column_tzs, external_data)
    177     response = self.command(query,
    178                             parameters=query_context.parameters,
    179                             settings=query_context.settings,
    180                             external_data=query_context.external_data)
    181     return QueryResult([response] if isinstance(response, list) else [[response]])
--> 182 return self._query_with_context(query_context)

File ~/Dev/streamlit-apps/.venv/lib/python3.10/site-packages/clickhouse_connect/driver/httpclient.py:217, in HttpClient._query_with_context(self, context)
    215 byte_source = RespBuffCls(ResponseSource(response))  # pylint: disable=not-callable
    216 context.set_response_tz(self._check_tz_change(response.headers.get('X-ClickHouse-Timezone')))
--> 217 query_result = self._transform.parse_response(byte_source, context)
    218 if 'X-ClickHouse-Summary' in response.headers:
    219     try:

File ~/Dev/streamlit-apps/.venv/lib/python3.10/site-packages/clickhouse_connect/driver/transform.py:62, in NativeTransform.parse_response(source, context)
     59     block_num += 1
     60     return result_block
---> 62 first_block = get_block()
     63 if first_block is None:
     64     return NumpyResult() if context.use_numpy else QueryResult([])

File ~/Dev/streamlit-apps/.venv/lib/python3.10/site-packages/clickhouse_connect/driver/transform.py:57, in NativeTransform.parse_response.<locals>.get_block()
     55         if error_start != -1:
     56             message = message[error_start:]
---> 57         raise StreamFailureError(message) from None
     58     raise
     59 block_num += 1

StreamFailureError: checkUInt8
```

### Steps to reproduce
1. Run code
2. See error

### Code example
```python
import clickhouse_connect

client = clickhouse_connect.get_client(
    host='HOST', 
    port=8443, 
    username='USER', 
    password='PASS,
    database='DB',    
)

client.query('SELECT 1 as check')
```

#### Environment
* clickhouse-connect version: 0.5.24
* Python version: 3.10.11
* Operating system: Mac OS

#### ClickHouse server
* ClickHouse Server version: 23.2.4
* ClickHouse Server non-default settings, if any: 
```
{
	"name": "max_query_size",
	"value": "10485760"
},
{
	"name": "use_uncompressed_cache",
	"value": "0"
},
{
	"name": "load_balancing",
	"value": "random"
},
{
	"name": "log_queries",
	"value": "1"
},
{
	"name": "join_use_nulls",
	"value": "1"
},
{
	"name": "any_join_distinct_right_table_keys",
	"value": "1"
},
{
	"name": "joined_subquery_requires_alias",
	"value": "0"
},
{
	"name": "max_memory_usage",
	"value": "128849018880"
},
{
	"name": "allow_experimental_object_type",
	"value": "1"
},
{
	"name": "allow_experimental_query_cache",
	"value": "1"
}
```

**Comment by genzgd** [comment]
A StreamFailureError happens when clickhouse-connect does not receive all of the data expected in a block from ClickHouse.  It may be that there is some incompatibility with this kind of stream processing and chproxy.  I assume this error happens every time?

Are there any errors in the chproxy or ClickHouse server logs?

Can you try `compress=False` in the `create_client` call?  Perhaps there's an issue with chproxy and the default `zstd` compression?

**Comment by denisov-vlad** [comment]
> I assume this error happens every time?

Yes, but I only tested this simple query.

> Are there any errors in the chproxy or ClickHouse server logs?

No, I didn't see any errors both in ch and chproxy logs.

> Can you try compress=False in the create_client call? Perhaps there's an issue with chproxy and the default zstd compression?

I've found some issues about compression and have tested already both with `False` and `zstd` options for the `compress` argument.

**Comment by genzgd** [comment]
This is actually a result of chproxy not passing the `client_protocol_version` version with the proxied request.  It is not included in the allowed headers -- see the allowed headers [here](https://github.com/ContentSquare/chproxy/blob/master/scope.go#L321-L346).  Because the ClickHouse server does not see the protocol version, it sends back data using the "default" protocol version of 0 that does not include the 8 byte prefix that `clickhouse-connect` expect to see (and throw away).

A temporary workaround is to add the line

```python
client.protocol_version = 0
```

after creating the client but before sending any queries.

I'll work on a cleaner fix in the next release.  I'll also open an issue in the `chproxy` project.

Unfortunately the chproxy approach of stripping most headers also breaks attempts to send specific settings (except for that limited list) with each query.

**Comment by genzgd** [comment]
This should be fixed in the 0.5.25 release.  Thanks for the report!
